### PR TITLE
SHS-5630: Editors can more clearly see what a change to media will affect

### PR DESCRIPTION
**Pull request by codechefmarc**
# READY FOR REVIEW

# Summary
- Changes the message for the number of items to include a link to the usage page to show which media items would be affected.

# Review By (Date)
- 8/19/2024

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- This will affect every site, however the change is minimal as it only affects back-end admin when editing a piece of media.

# Urgency
- Low

# Review Tasks

## Setup tasks and/or behavior to test

1. Testing steps and testing site located here: https://github.com/SU-HSDO/suhumsci/pull/1583

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?
- No

## Front End Validation
- [ ] Design is approved by @ user? (Not sure)
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: composer.json
Comment (joegl) [inline composer.json]: This will need to be changed back to `su-sws/` to merge.

**Comment by joegl** [comment]
Adding @pookmish to review.

File: docs/architecture/decisions/0001-media-usage-warning-link.md
Comment (pookmish) [inline docs/architecture/decisions/0001-media-usage-warning-link.md]: Remove this file.

File: stanford_media.module
Comment (pookmish) [inline stanford_media.module]: Use URL object from `$media->toUrl('usage')`

File: stanford_media.module
Comment (pookmish) [inline stanford_media.module]: Just use `t()`

File: stanford_media.module
Comment (pookmish) [inline stanford_media.module]: Our current convention is snake case, not camel case.

File: stanford_media.module
Comment (pookmish) [reply stanford_media.module]: Change to snake case

**Review by pookmish — CHANGES_REQUESTED** [review]
I haven't tested yet, but there are some things to address at this time.

File: docs/architecture/decisions/0001-media-usage-warning-link.md
Comment (codechefmarc) [reply docs/architecture/decisions/0001-media-usage-warning-link.md]: I will remove. But wanted to make a note that SWS did request that we start adding these ADR documents.

File: composer.json
Comment (codechefmarc) [reply composer.json]: I did not change this yet, as we still need to be able to test on `suhumsci` but feel free to switch this before merge.

File: docs/architecture/decisions/0001-media-usage-warning-link.md
Comment (codechefmarc) [reply docs/architecture/decisions/0001-media-usage-warning-link.md]: Removed.

File: stanford_media.module
Comment (codechefmarc) [reply stanford_media.module]: Changed these.

File: stanford_media.module
Comment (codechefmarc) [reply stanford_media.module]: Changed.

File: stanford_media.module
Comment (codechefmarc) [reply stanford_media.module]: Changed.

**Comment by codechefmarc** [comment]
Made all changes except for the `composer.json` name change - I'll let you do that prior to merge.

File: stanford_media.module
Comment (pookmish) [inline stanford_media.module]: `\Drupal\Core\Link::fromTextAndUrl()` does this for you.


```suggestion
    $formatted_link = Link::fromTextAndUrl($plural_text, $media->toUrl('usage'));
```

**Review by pookmish — CHANGES_REQUESTED** [review]

File: stanford_media.module
Comment (codechefmarc) [reply stanford_media.module]: Ok, fixed.

**Review by pookmish — APPROVED** [review]
I'm good with this now once the composer.json is reverted

**Comment by codechefmarc** [comment]
Composer reverted.

**Pull request merged**
